client/src/app/shared/images/global-icon.component.ts

```diff
@@ -86,7 +88,22 @@ export class GlobalIconComponent implements OnInit {
     )
   }
 
+  private getId (id: number | string = uniqueId()): string {
+    return lowerFirst(this.constructor.name) + `_${id}-`
+  }
+
+  private getUniquelyPrefixedSVG (svgStr: String) {
+    return SVGInlineLoader.getExtractedSVG(
+      svgStr,
+      {
+        idPrefix: this.getId(),
```

Comment:
Do we really need to have an id for SVGs?

Reply:
Yes, and anyway cleaning up all svg in advance is going to be a major pain.

Reply:
> Yes

Why?

>  cleaning up all svg in advance would be a major PITA

doing this at runtime is not the appropriate solution

Reply:
Apparently, some SVG elements will have no effect without an id. Though I have yet to find an icon we use where that occurs, I deemed safer to avoid the situation altogether.

source: https://oreillymedia.github.io/Using_SVG/guide/markup.html

> > cleaning up all svg in advance would be a major PITA
>
> doing this at runtime is not the appropriate solution

The I guess we go manual.

Reply:
> Then I guess we go manual.

We can also find/create a tool to cleanup svg and generate IDs, so we don't add dependencies at runtime and avoid to do things manually

Reply:
I found https://jakearchibald.github.io/svgomg/ (based on cli utility `svgo`). It fits the bill.